ch11 uptimer tests

When I run the ch11 tests I seem to get an error with the click style related tests.  Do you think is powershell related?

Windows 10
Python 3.8.5


```powershell
===================================================================== test session starts =====================================================================
platform win32 -- Python 3.8.5, pytest-6.2.4, py-1.10.0, pluggy-0.13.1 -- c:\users\bmcneil\git\courses\modern-python-projects\uptimer\.venv\scripts\python.exe
cachedir: .pytest_cache
rootdir: C:\Users\bmcneil\git\courses\modern-python-projects\uptimer
plugins: mock-3.6.1
collected 9 items

tests/test_uptimer.py::test_version PASSED                                                                                                               [ 11%]
tests/test_uptimer.py::test_check_url PASSED                                                                                                             [ 22%]
tests/test_uptimer.py::test_colorize_status PASSED                                                                                                       [ 33%]
tests/test_uptimer.py::test_check_one_url[200-green] FAILED                                                                                              [ 44%]
tests/test_uptimer.py::test_check_one_url[304-yellow] FAILED                                                                                             [ 55%]
tests/test_uptimer.py::test_check_one_url[404-bright_red] FAILED                                                                                         [ 66%]
tests/test_uptimer.py::test_check_one_url[500-red] FAILED                                                                                                [ 77%]
tests/test_uptimer.py::test_check_one_url[1-magenta] FAILED                                                                                              [ 88%]
tests/test_uptimer.py::test_check_multiple_urls FAILED                                                                                                   [100%]

========================================================================== FAILURES ===========================================================================
________________________________________________________________ test_check_one_url[200-green] ________________________________________________________________

mocker = <pytest_mock.plugin.MockerFixture object at 0x0000014DF5C5CE80>, code = 200, color = 'green'

    @pytest.mark.parametrize(
        "code,color",
        [
            (200, "green"),
            (304, "yellow"),
            (404, "bright_red"),
            (500, "red"),
            (1, "magenta"),
        ],
    )
    def test_check_one_url(mocker, code, color):
        mocker.patch("requests.head", return_value=mock_response_object(code))

        runner = CliRunner()
        result = runner.invoke(check, ["dummyurl"], color=True)

        expected_message = click.style(f"dummyurl -> {code}", fg=color)
>       assert result.output == f"{expected_message}\n"
E       AssertionError: assert 'dummyurl -> 200\n' == '\x1b[32mdummyurl -> 200\x1b[0m\n'
E         - dummyurl -> 200
E         ? -----               ----
E         + dummyurl -> 200

tests\test_uptimer.py:58: AssertionError
_______________________________________________________________ test_check_one_url[304-yellow] ________________________________________________________________

mocker = <pytest_mock.plugin.MockerFixture object at 0x0000014DF5C668E0>, code = 304, color = 'yellow'

    @pytest.mark.parametrize(
        "code,color",
        [
            (200, "green"),
            (304, "yellow"),
            (404, "bright_red"),
            (500, "red"),
            (1, "magenta"),
        ],
    )
    def test_check_one_url(mocker, code, color):
        mocker.patch("requests.head", return_value=mock_response_object(code))

        runner = CliRunner()
        result = runner.invoke(check, ["dummyurl"], color=True)

        expected_message = click.style(f"dummyurl -> {code}", fg=color)
>       assert result.output == f"{expected_message}\n"
E       AssertionError: assert 'dummyurl -> 304\n' == '\x1b[33mdummyurl -> 304\x1b[0m\n'
E         - dummyurl -> 304
E         ? -----               ----
E         + dummyurl -> 304

tests\test_uptimer.py:58: AssertionError
_____________________________________________________________ test_check_one_url[404-bright_red] ______________________________________________________________

mocker = <pytest_mock.plugin.MockerFixture object at 0x0000014DF5CBD5E0>, code = 404, color = 'bright_red'

    @pytest.mark.parametrize(
        "code,color",
        [
            (200, "green"),
            (304, "yellow"),
            (404, "bright_red"),
            (500, "red"),
            (1, "magenta"),
        ],
    )
    def test_check_one_url(mocker, code, color):
        mocker.patch("requests.head", return_value=mock_response_object(code))

        runner = CliRunner()
        result = runner.invoke(check, ["dummyurl"], color=True)

        expected_message = click.style(f"dummyurl -> {code}", fg=color)
>       assert result.output == f"{expected_message}\n"
E       AssertionError: assert 'dummyurl -> 404\n' == '\x1b[91mdummyurl -> 404\x1b[0m\n'
E         - dummyurl -> 404
E         ? -----               ----
E         + dummyurl -> 404

tests\test_uptimer.py:58: AssertionError
_________________________________________________________________ test_check_one_url[500-red] _________________________________________________________________

mocker = <pytest_mock.plugin.MockerFixture object at 0x0000014DF5CF2040>, code = 500, color = 'red'

    @pytest.mark.parametrize(
        "code,color",
        [
            (200, "green"),
            (304, "yellow"),
            (404, "bright_red"),
            (500, "red"),
            (1, "magenta"),
        ],
    )
    def test_check_one_url(mocker, code, color):
        mocker.patch("requests.head", return_value=mock_response_object(code))

        runner = CliRunner()
        result = runner.invoke(check, ["dummyurl"], color=True)

        expected_message = click.style(f"dummyurl -> {code}", fg=color)
>       assert result.output == f"{expected_message}\n"
E       AssertionError: assert 'dummyurl -> 500\n' == '\x1b[31mdummyurl -> 500\x1b[0m\n'
E         - dummyurl -> 500
E         ? -----               ----
E         + dummyurl -> 500

tests\test_uptimer.py:58: AssertionError
________________________________________________________________ test_check_one_url[1-magenta] ________________________________________________________________

mocker = <pytest_mock.plugin.MockerFixture object at 0x0000014DF5CCEB80>, code = 1, color = 'magenta'

    @pytest.mark.parametrize(
        "code,color",
        [
            (200, "green"),
            (304, "yellow"),
            (404, "bright_red"),
            (500, "red"),
            (1, "magenta"),
        ],
    )
    def test_check_one_url(mocker, code, color):
        mocker.patch("requests.head", return_value=mock_response_object(code))

        runner = CliRunner()
        result = runner.invoke(check, ["dummyurl"], color=True)

        expected_message = click.style(f"dummyurl -> {code}", fg=color)
>       assert result.output == f"{expected_message}\n"
E       AssertionError: assert 'dummyurl -> 1\n' == '\x1b[35mdummyurl -> 1\x1b[0m\n'
E         - dummyurl -> 1
E         ? -----             ----
E         + dummyurl -> 1

tests\test_uptimer.py:58: AssertionError
__________________________________________________________________ test_check_multiple_urls ___________________________________________________________________

mocker = <pytest_mock.plugin.MockerFixture object at 0x0000014DF5D00700>

    def test_check_multiple_urls(mocker):
        mocker.patch(
            "requests.head",
            side_effect=[mock_response_object(200), mock_response_object(500)],
        )

        runner = CliRunner()
        result = runner.invoke(check, ["dummyurl1", "dummyurl2"], color=True)

        expected_message1 = click.style("dummyurl1 -> 200", fg="green")
        expected_message2 = click.style("dummyurl2 -> 500", fg="red")
>       assert result.output == f"{expected_message1}\n{expected_message2}\n"
E       AssertionError: assert 'dummyurl1 -> 200\ndummyurl2 -> 500\n' == '\x1b[32mdummyurl1 -> 200\x1b[0m\n\x1b[31mdummyurl2 -> 500\x1b[0m\n'
E         - dummyurl1 -> 200
E         ? -----                ----
E         + dummyurl1 -> 200
E         - dummyurl2 -> 500
E         ? -----                ----
E         + dummyurl2 -> 500

tests\test_uptimer.py:72: AssertionError
=================================================================== short test summary info ===================================================================
FAILED tests/test_uptimer.py::test_check_one_url[200-green] - AssertionError: assert 'dummyurl -> 200\n' == '\x1b[32mdummyurl -> 200\x1b[0m\n'
FAILED tests/test_uptimer.py::test_check_one_url[304-yellow] - AssertionError: assert 'dummyurl -> 304\n' == '\x1b[33mdummyurl -> 304\x1b[0m\n'
FAILED tests/test_uptimer.py::test_check_one_url[404-bright_red] - AssertionError: assert 'dummyurl -> 404\n' == '\x1b[91mdummyurl -> 404\x1b[0m\n'
FAILED tests/test_uptimer.py::test_check_one_url[500-red] - AssertionError: assert 'dummyurl -> 500\n' == '\x1b[31mdummyurl -> 500\x1b[0m\n'
FAILED tests/test_uptimer.py::test_check_one_url[1-magenta] - AssertionError: assert 'dummyurl -> 1\n' == '\x1b[35mdummyurl -> 1\x1b[0m\n'
FAILED tests/test_uptimer.py::test_check_multiple_urls - AssertionError: assert 'dummyurl1 -> 200\ndummyurl2 -> 500\n' == '\x1b[32mdummyurl1 -> 200\x1b[0m\n\...
================================================================= 6 failed, 3 passed in 0.57s =================================================================
```

--- a/ch11-cli/uptimer/tests/test_uptimer.py
+++ b/ch11-cli/uptimer/tests/test_uptimer.py
@@ -1,6 +1,7 @@
 import click
 import pytest
 import requests
+import sys
 from click.testing import CliRunner
 from uptimer import __version__
 from uptimer.uptimer import check, check_url, colorize_status
@@ -38,6 +39,9 @@ def test_colorize_status(mocker):
     click.secho.assert_called_with(f"{url} -> {status}", fg="green")
 
 
+@pytest.mark.skipif(
+    sys.platform == "win32", reason="Testing colorized output doesn't work on Windows"
+)
 @pytest.mark.parametrize(
     "code,color",
     [
@@ -58,6 +62,9 @@ def test_check_one_url(mocker, code, color):
     assert result.output == f"{expected_message}\n"
 
 
+@pytest.mark.skipif(
+    sys.platform == "win32", reason="Testing colorized output doesn't work on Windows"
+)
 def test_check_multiple_urls(mocker):
     mocker.patch(
         "requests.head",

